tc/core/halide2isl.cc

```diff
@@ -548,20 +551,16 @@ std::vector<Reduction> findReductions(const Stmt& s) {
       if (isReductionInit(op)) {
         reductions[op->name].init = op;
         initVars[op->name] = vars;
-      } else if (isReductionUpdate(op)) {
-        if (isValidReductionUpdate(op)) {
-          auto& p = reductions[op->name];
-          CHECK(p.init.defined())
-              << "Missing reduction init or (unsupported) multiple updates";
-          CHECK(!p.update.defined())
-              << "Multiple reduction updates not yet implemented";
-          p.update = op;
-          auto n = initVars[op->name].size();
-          p.dims.resize(vars.size() - n);
-          std::iota(p.dims.begin(), p.dims.end(), n);
-        } else {
-          reductions.erase(op->name);
-        }
+      } else if (isValidReductionUpdate(op)) {
+        auto& p = reductions[op->name];
+        CHECK(p.init.defined())
+            << "Missing reduction init or (unsupported) multiple updates";
```

Comment:
The error message should be updated.
The original code could end up removing the init in a previous update,
but the new code no longer removes reductions.